Smt/Constants.lean

```diff
@@ -0,0 +1,20 @@
+import Lean
+
+import Smt.Solver
+import Smt.Term
+
+namespace Smt
+namespace Constants
+
+open Smt.Term
+open Smt.Solver
+open Lean
+
+def natMinus : Expr := mkConst (Name.mkSimple "natMinus")
+
+def defNatMinus : String :=
+   genDefineFun "natMinus" [("x", `"Int"), ("y", `"Int")] (`"Int") $
+    `"ite" • (`"<" • `"x" • `"y") • ``"0" • (`"-" • `"x" • `"y")
```

Comment:
Yeah, that would be better, I agree. But then, what should be the return type of markMinus? We would have to return a list of functions (Solver -> Solver)? I guess this would create problems if we try to erase duplicates in the return list, which is something I wanted to do to avoid repeating definitions... can you think of any other way to pass the information of which functions should be define to the function that called markMinus?